engine.py

```diff
@@ -108,6 +116,57 @@ def post_app_init(self):
             except:
                 traceback.print_exc()
 
+    def validate_context(self, tank, context):
+        '''
+        Given the context provided at initialisation, if we're in a project context only, forces the user to select
+        a proper task context using rdokatana.taskChooser.taskChooser.TaskChooser.
+        :param tank: a Tank object instance.
+        :type tank: :class:`sgtk.tank.Tank`
+        :param context: the current context.
+        :type context: :class:`tank.context.Context`
+        :return: a proper working context.
+        :rtype: :class:`tank.context.Context`
+        '''
+        if context.project:
+            if not context.entity:
+                task = self.select_task_ui(context)
+                newContext = tank.context_from_entity('Task', task['id'])
+                return newContext
+            elif context.entity and not context.task:
+                filters = [
+                    ['project', 'is', context.project],
+                    ['entity', 'is', context.entity],
+                ]
+                task = None
+                if context.entity['type'] == 'Shot':
+                    filters.append(['content', 'is', 'Lighting'])
+                    task = tank.shotgun.find_one("Task", filters)
+                elif context.entity['type'] == 'Asset':
+                    filters.append(['content', 'is', 'Shading'])
+                    task = tank.shotgun.find_one("Task", filters)
+                if task:
+                    newContext = tank.context_from_entity('Task', task['id'])
+                    return newContext
+                else:
+                    task = self.select_task_ui(context)
+                    newContext = tank.context_from_entity('Task', task['id'])
+                    return newContext
```

Comment:
That's a request from the "client".
We had three options:
1/ You can have a project context only, but in this case, when you're trying to publish something, instead of showing the Publish UI, it would either not create the tab or create a tab with an error message saying your context is not set properly.
2/ Select a default context, Peter wanted it to be a "library" shot. I don't really like this option personnally.
3/ Enforce the context. After discussing, we chose the third option. I don't really see why this is different than nuke for example, where you enforce the choice of a shot at launch time.